synapse/config/experimental.py

```diff
@@ -14,10 +14,25 @@
 
 from typing import Any
 
+import attr
+
 from synapse.config._base import Config
 from synapse.types import JsonDict
 
 
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class MSC3866Config:
```

Comment:
Looks like we're only doing it in http handlers. Also I'm not sure I see the point of validation to this extent for experimental, if-you-use-this-without-knowing-exactly-what-you're-doing-you're-on-your-own options.

Reply:
We're free to use it wherever we want to. The original suggestion was to use it for config and I even prototyped this: https://github.com/matrix-org/synapse/issues/12651#issuecomment-1133940984

We don't have many examples of using Pydantic yet so I wouldn't force it on anyone. We would probably need some glue code for config too?

But I think it is pretty good at what it does, gives better error messages earlier and gives us an opportunity to propagate more types through Synapse.

Reply:
No strong opinion but presumably this won't always be an experimental option, at which point it will just get renamed into something unexperimental? So I don't really buy that we should do things less properly for experimental options, but it's only a minor thought